Converted Plugins.md to single files (jekyll)

Falke-Design opened the pull request:
Based on #7804. Around 489 plugin files.

You can look into the convert script [here](https://github.com/Falke-Design/Leaflet/commit/f00794cc5d36c443afd004afe9756aeac97a674d)

![grafik](https://user-images.githubusercontent.com/19800037/143076779-a29e2a45-2d33-4ac9-bb72-a4ae9272e776.png)

Missing:
- Order of plugins (most used above, outdated ones below)
- ~~`compatible-v0: false` check~~
- ~~Remove [Demo](#) from text~~

FYI @IvanSanchez @mourner

IvanSanchez commented on docs/_plugins/angular-leaflet-directive.md:
```diff
@@ -0,0 +1,12 @@
+---
+name: Angular Leaflet directive
+category: frameworks-&-build-systems
```
I would rather not have `&` in the category names. I don't trust all YAML parsers will play nicely with that.

Please modify your script (at https://github.com/Falke-Design/Leaflet/commit/f00794cc5d36c443afd004afe9756aeac97a674d#diff-41c60cd16ab76d71244c42809a6f886d662c8c676b3de8f2fe1e5f61ccafee6eR128 ) to provide category names without `&`s.

IvanSanchez commented on docs/_plugins/abp-usermap-mybb.md:
```diff
@@ -0,0 +1,12 @@
+author: CrazyCat
+author-url: https://gitlab.com/AnoBug
+demo: 
+compatible-v0: false
```
Best to leave the `compatible` fields empty for the time being. Can be checked and changed on a one-by-one basis afterwards.

Falke-Design replied:
I set only `compatible-v0` to empty and went through all plugins to check if some of them are only compatible with v0.

IvanSanchez commented on docs/plugins-database.md:
```diff
@@ -0,0 +1,386 @@
+---
```
Rename back to `plugins.md`. I changed the name in #7804 to avoid deleting the original plugins list,but that's no longer a concern here.

IvanSanchez reviewed (approved):
Looks good!

Falke-Design commented:
Before merging I would like to update the [corrupt URLs](https://florianbischof.net/leaflet/links.html) like in #7801 described.

What should we do with the [open PRs](https://github.com/Leaflet/Leaflet/pulls?q=is%3Aopen+is%3Apr+label%3Aplugins)? Many of them have only merge conflicts or useing blanks instead of tabs. I can insert them into this PR but then the users will not be listed as contributors, is this a problem?

IvanSanchez commented:
> Before merging I would like to update the [corrupt URLs](https://florianbischof.net/leaflet/links.html) like in #7801 described.

I don't have a strong opinion. Perhaps run your script against an updated plugins list?

Oh, BTW, I haven't tested it but it might be possible to put the individual `.md` in directories - so instead of 500 files it should be possible to have 20 directories (one per category) and a more manageable amount of files per directory.

> What should we do with the [open PRs](https://github.com/Leaflet/Leaflet/pulls?q=is%3Aopen+is%3Apr+label%3Aplugins)? I can insert them into this PR but then the users will not be listed as contributors, is this a problem?

It's a very minor problem IMO, given the circumstances. The priority is on cleaning up the list and prevent merge conflicts down the line.

jonkoops commented:
This is a great addition!

Falke-Design commented:
@IvanSanchez @jonkoops please check the [template commit](https://github.com/Leaflet/Leaflet/pull/7805/commits/735289bcb59eac743ee68dee5467b7a516763a23). Is the english / phrasing ok?

- [x] Updated links
- [x] Moved the plugins to category directories
- [x] Updated describtion and added `template.md`
- [x] Added [open PRs](https://github.com/Leaflet/Leaflet/pulls?q=is%3Aopen+is%3Apr+label%3Aplugins)

Some other related issues (for documentation): #5387, #3240

In my eyes this PR is ready to merge

jonkoops commented on docs/_plugins/template.md:
```diff
@@ -0,0 +1,12 @@
+---
+name: requried
+category: requried (same name as directory)
+repo: requried
+author: requried
```
This should be 'required'

jonkoops commented on docs/_plugins/template.md:
```diff
@@ -0,0 +1,12 @@
+compatible-v1: true
+---
+
+Describe here your plugin. Keep it short and check out the https://github.com/Leaflet/Leaflet/blob/master/PLUGIN-GUIDE.md.
```
You could add a markdown link perhaps?

```
[plugin guide](https://github.com/Leaflet/Leaflet/blob/master/PLUGIN-GUIDE.md)
```

IvanSanchez replied:
Or mention that markdown can be used if needed.

IvanSanchez commented on docs/_plugins/template.md:
```diff
@@ -0,0 +1,12 @@
+---
+name: requried
+category: requried (same name as directory)
```
Typo here!!

IvanSanchez reviewed (approved):
I think this is ready.

IvanSanchez commented:
@Falke-Design Thanks a lot for doing the bulk of the work converting the table to individual files :smile:

brunob commented:
I was thinking of doing a pass on all plugins in order to mark the ones that seems unmaintained or "dead" and discovering this PR, and i just have to say that this is awesome ! Thx for that :)

Anyway, i think i'll push some PR to update the `compatible-vX` on some plugins.
